plugin.video.retrospect/resources/libs/addonsettings.py

```diff
@@ -0,0 +1,1430 @@
+
+    @staticmethod
+    def hide_fanart():
+        """ Should we hide Fanart?
+
+        :return: Yes or No
+        :type: bool
+
+        """
+
+        return AddonSettings.store(KODI).get_boolean_setting("hide_fanart")
+
+    @staticmethod
+    def hide_drm_items():
+        """ Returns whether or not to hide DRM protected items.
+
+        :return: True/False
+        :type: bool
+
+        """
+
+        return AddonSettings.store(KODI).get_boolean_setting("hide_drm")
+
+    @staticmethod
+    def hide_premium_items():
+        """ Returns whether or not to hide Premium/Paid items.
+
+        :return: True/False
+        :type: bool
+
+        """
+
+        return AddonSettings.store(KODI).get_boolean_setting("hide_premium")
+
+    @staticmethod
+    def hide_restricted_folders():
+        """ Should we hide restricted folders?
+
+        :return: Indaction if the restricted folders should be hidden
+        :rtype: bool
+
+        """
+
+        values = [True, False]
+        value = AddonSettings.store(KODI).get_integer_setting("hide_types", default=0)
+        return values[value]
+
+    @staticmethod
+    def hide_first_time_messages():
+        """ Indication if we should show or hide the channel initialization messages
+
+        :return: returns true if the first time messages should be shown.
+        :rtype: bool
+
+        """
+
+        return AddonSettings.store(KODI).\
+            get_boolean_setting("hide_first_time_message", default=False)
+    #endregion
+
+    #region Geo and region stuff
+    @staticmethod
+    def get_available_countries(as_string=False, as_country_codes=False):
+        """ returns the all available ProxyGroupId's in order. The countries are:
+
+             :param bool as_country_codes:  Returns alls the actual country codes values.
+             :param bool as_string:         Returns the translation ID for all the possible country
+                                            codes as strings.
+
+             :return: List[str] A list of either country codes or translation ID's
+
+             * other - Other languages
+             * uk    - United Kingdom
+             * nl    - The Netherlands
+             * se    - Sweden
+             * no    - Norway
+             * de    - Germany
+             * be    - Belgium
+             * ee    - Estonia
+             * lt    - Lithuani
+             * lv    - Latvia
+             * dk    - Danish
+
+        """
+
+        proxy_ids = [30025, 30300, 30301, 30307, 30302, 30305, 30309, 30306, 30308, 30303, 30304, 30310]
+        proxy_codes = [None, "other", "nl", "uk", "se", "no", "de", "be", "ee", "lt", "lv", "dk"]
+
+        if as_string:
+            return [str(i) for i in proxy_ids]
+
+        if as_country_codes:
+            return proxy_codes
+
+        return proxy_ids
+
+    @staticmethod
+    def hide_geo_locked_items_for_location(channel_region, value_only=False):
```

Comment:
not allowed

Reply:
This is not what you might think. Some broadcast companies have streams that cannot be watched outside their Geo region. However, on their site those streams are listed anyways but they might show an icon to indicate Geo locking. 

This setting just allows the user to hide those items that will not playback in their Geo region, so the listing they get within Kodi only contain the streams that will play taking their geo location into account. 

E.g.: If a show is Geo locked for The Netherlands, I would see the show in the listing, as I have configured the Geo region for Retrospect to be The Netherlands. But for me, a Swedish Geo locked show would not appear in my listing as the region does not match my configured one.

Reply:
ignore